### ISSUE TITLE
Enhance test suite to run test against running sssd and LDAP server

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#641]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/637
- Created at 2010-10-01 18:33:44 by [sbose](https://pagure.io/user/sbose)
- Closed at 2020-04-20 13:36:11 as Fixed
- Assigned to nobody
---

It would be very nice if we can run tests during 'make check' which perform test against a running sssd anbd a running LDAP server (or stopped LDAP server to simulate the offline case). It would be even nicer if the tests can be run as non-privileged user, because this would allow us to run the tests even in build farms like koji or the OpenSUSE build service.

It is possible to configure and run OpenLDAP so that it can be used inside the test suite. I assume it would be possible for 389ds, too.

sssd uses a couple paths hard-coded at configure time and has also some tests if files are owned by root or processes are run with root privileges. Some research and discussion is needed about what would be the best way to run sssd in a 'test mode'.

### Comments

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2010-10-14 15:20:24](https://pagure.io/SSSD/sssd/issue/637#220409)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.6.0

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2011-02-07 14:55:50](https://pagure.io/SSSD/sssd/issue/637#220410)**

Fields changed

coverity:  => 
milestone: SSSD 1.6.0 => SSSD Deferred
upgrade:  => 0

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2012-01-19 03:19:37](https://pagure.io/SSSD/sssd/issue/637#220411)**

Fields changed

rhbz:  => 0

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2013-07-08 07:09:59](https://pagure.io/SSSD/sssd/issue/637#220412)**

This would be much simpler with Andreas' socket/nss/uid wrappers.

blockedby:  => 
blocking:  => 
changelog:  => 
design:  => 
design_review:  => 0
feature_milestone:  => 
fedora_test_page:  => 
milestone: SSSD Deferred => NEEDS_TRIAGE
patch:  => 0
review:  => 0
selected:  =>

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2013-07-09 11:10:17](https://pagure.io/SSSD/sssd/issue/637#220413)**

http://cwrap.org/

cc:  => asn@redhat.com

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2013-07-11 15:59:45](https://pagure.io/SSSD/sssd/issue/637#220414)**

Fields changed

milestone: NEEDS_TRIAGE => Interim Bucket
type: enhancement => task

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2013-07-30 12:54:43](https://pagure.io/SSSD/sssd/issue/637#220415)**

Fields changed

milestone: Interim Bucket => SSSD 1.12 beta

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2014-02-19 11:51:39](https://pagure.io/SSSD/sssd/issue/637#220416)**

Fields changed

blocking:  => #1923

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2014-05-28 14:41:21](https://pagure.io/SSSD/sssd/issue/637#220417)**

There was no time to work on this during the 1.12 cycle, sorry. Re-proposing to 1.13

review: 0 => 1

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2014-05-30 16:01:38](https://pagure.io/SSSD/sssd/issue/637#220418)**

Fields changed

milestone: SSSD 1.12 beta => SSSD 1.13 beta

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2014-09-26 18:49:56](https://pagure.io/SSSD/sssd/issue/637#220419)**

Fields changed

mark:  => 0

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2015-02-10 16:17:21](https://pagure.io/SSSD/sssd/issue/637#220420)**

Nikolai is actively working on this.

milestone: SSSD 1.13 beta => SSSD Continuous integration
owner: somebody => nkondrashov

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2017-02-24 14:21:54](https://pagure.io/SSSD/sssd/issue/637#220421)**

**Metadata Update from @sbose**:
- Issue marked as blocked by: #1923
- Issue set to the milestone: SSSD Continuous integration

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2020-04-20 13:36:13](https://pagure.io/SSSD/sssd/issue/637#642978)**

This basically talks about 'make intgcheck' which was done years ago. Closing.

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2020-04-20 13:36:14](https://pagure.io/SSSD/sssd/issue/637#642979)**

**Metadata Update from @pbrezina**:
- Custom field blocking reset (from #1923)
- Custom field design_review reset (from 0)
- Custom field mark reset (from 0)
- Custom field patch reset (from 0)
- Custom field review adjusted to on (was: 1)
- Custom field testsupdated reset (from 0)
- Issue close_status updated to: Fixed
- Issue status updated to: Closed (was: Open)
